nvflare/app_common/widgets/intime_model_selector.py

```diff
@@ -109,8 +123,25 @@ def _before_accept(self, fl_ctx: FLContext):
             self.log_warning(fl_ctx, f"validation metric not existing in {client_name}")
             return False
 
+        if self.weigh_by_local_iter:
+            n_iter = dxo.get_meta_prop(MetaKey.NUM_STEPS_CURRENT_ROUND, 1.0)
+        else:
+            n_iter = 1.0
+
+        if self.tb_summary:
+            if client_name not in self._writers:
+                app_root = fl_ctx.get_prop(FLContextKey.APP_ROOT)
+                self._writers[client_name] = tb.SummaryWriter(os.path.join(app_root, client_name))
+                self.log_info(fl_ctx, f"Attempting to write TensorBoard events for {client_name} to {self.app_root}")
+
         # select key metric if dictionary of metrics is provided
         if isinstance(validation_metric, dict):
+            if self.tb_summary:
+                for _metric, _value in validation_metric.items():
+                    self._writers[client_name].add_scalar(
+                        f"{_metric}_{client_name}", _value, int(current_round * n_iter)
+                    )
+
```

Comment:
I think these log-related logic we should create a new "widgets" that implements "handle_event" and in "before_accept" event that new widgets log these stuffs.

I think we should not combine the TB summary writer inside model selector.

Reply:
That's a good suggestion. Maybe this new widget can use analytics Receivers, so it will show up in the MLexp trackers (MLFLow, Tensorboard, or W&B) based on the configuration. CC @chesterxgchen .

Reply:
I removed the TB logging from this PR. Let's create another Widget for this purpose.